### DISCUSSION TITLE
How to change the value of publicPath at runtime?

### DISCUSSION THREAD

**Discussion by wtagain** [nuxt/nuxt#9293] [Questions (Nuxt 2)]
Document: https://nuxtjs.org/docs/2.x/configuration-glossary/configuration-build#publicpath

```
In Nuxt 2.15+, changing the value of this property at runtime will override the configuration of an app that has already been built.
```

The question is how to changing the value of this property at runtime? 

Now, I tried to set a new publicPath when in hooks-render-resourcesLoaded
nuxt.config.ts
```
hooks: {
  render: {
    resourcesLoaded(resources) {
      // If I want to change the publichPath dynamically, like the main cdn isn't available.
      resources.clientManifest.publicPath = process.env.CDN_URL_BACKUP
    }
  }
}
```
Is there any other solution?

**Comment by zcs02468** [comment]
I also encountered the same problem. I needed to dynamically set different publicPath paths according to the IP obtained in the page access request req
